lib/handler/connect.c

```diff
@@ -147,34 +215,44 @@ static void on_getaddr(h2o_hostinfo_getaddr_req_t *getaddr_req, const char *errs
         start_connect(creq);
     } else {
         assert(res->ai_socktype == SOCK_DGRAM);
-        h2o_tunnel_t *tunnel = h2o_open_udp_tunnel_from_sa(creq->loop, res->ai_addr, res->ai_addrlen);
+        const char *err = NULL;
+        h2o_tunnel_t *tunnel = h2o_open_udp_tunnel_from_sa(creq->loop, res->ai_addr, res->ai_addrlen, &err);
         h2o_req_t *req = creq->src_req;
-        uint64_t timeout = creq->handler->config.io_timeout;
-        req->res.status = 200;
         h2o_timer_unlink(&creq->timeout);
-        req->establish_tunnel(req, tunnel, timeout);
+        if (tunnel != NULL) {
+	    req->res.status = 200;
+	    uint64_t timeout = creq->handler->config.io_timeout;
+	    req->establish_tunnel(req, tunnel, timeout);
+        } else {
+            h2o_req_log_error(req, "lib/handler/connect.c", "Failed to create downstream socket");
```

Comment:
I think we might want to emit `err` here too?

In principle, I tend to believe that the error that we send to the client should be a proper subset of what we log. This argument also applies to other occasions that we send the proxy-status header field.